hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestKeyManagerImpl.java

```diff
@@ -1351,6 +1353,80 @@ public void testGetFileStatusWithFakeDir() throws IOException {
     Assert.assertTrue(ozoneFileStatus.isFile());
   }
 
+  @Test
+  public void testGetFileStatusWithFakeDirFalsePositive() throws IOException {
```

Comment:
```suggestion
  private static Stream<Arguments> getFileStatusWithFakeDirFalsePositiveScenarios() {
    String dir0 = "dir0";
    String dir1 = "dir1";
    String dir2 = "dir2";
    String dir3 = "dir3";
    String keyName1 = dir1 + OZONE_URI_DELIMITER + "file1";
    String keyName2 = dir2 + OZONE_URI_DELIMITER + "file2";

    return Stream.of(
        Arguments.of("Case 1: volume1/bucket1/dir1/file1", // You can explain more here about the test case like input and out
            BUCKET_NAME,
            keyName1,
            Collections.singletonList(dir1),
            Arrays.asList(dir0, dir2, dir3)
        ),
        Arguments.of("Case 2: volume1/bucket2/dir2/file2",  // You can explain more here about the test case like input and out
            BUCKET_2_NAME,
            keyName2,
            Collections.singletonList(dir2),
            Arrays.asList(dir0, dir1, dir3)
        )
    );
  }

  @ParameterizedTest(name = "{0}")
  @MethodSource("getFileStatusWithFakeDirFalsePositiveScenarios")
  public void testGetFileStatusWithFakeDirFalsePositive(String description,
                                                        String bucketName,
                                                        String keyName,
                                                        List<String> expectedDirs,
                                                        List<String> notExpectedDirs)
      throws IOException {

    createFile(bucketName, keyName);

    Assert.assertNotNull(metadataManager.getKeyTable(getDefaultBucketLayout())
        .get(metadataManager.getOzoneKey(VOLUME_NAME, BUCKET_NAME, keyName)));
    // Assert volumeName, bucketName and keyName as well;

    for (String notExpectedDir: notExpectedDirs) {
      assertFileNotFound(BUCKET_NAME, notExpectedDir);
    }
    for (String expectedDir: expectedDirs) {
      assertIsDirectory(BUCKET_NAME, expectedDir);
    }
  }
```

I think it would be easier to read and understand the test scenarios if you change the test to parameterize in Junit-5. And it can be a single test instead of `testGetFileStatusWithFakeDirFalsePositive` and `testGetFileStatusWithFakeDirFalseNegative` as separate tests.

Reply:
This test class is still in Junit 4, let's not change it now?

Reply:
Junit-5 is already a dependency. You just have to change the import to junit-5. No change would be needed in the existing tests.

Reply:
Thanks, I have updated the test.

Reply:
Thanks for the upgrading to Junit-5 @kaijchen.

Can you please remove Junit-4 import for this class? And change `Assertions`.

Reply:
@hemantk-12 let's not mix refactor and bugfix

Reply:
Hi @adoroszlai , I agree that generally in a bug fix we should not do refactoring (because of backportability concern). Though in this case I'd say it is alright as the "refactoring" is in the UT.

I'm fine either way. :)

Reply:
Hi @adoroszlai,
As @smengcl , said I agree that it is not a good idea to mix bug fix and refactoring. But if you look at the original patch, these are the new tests which were added and I asked them to be in Junit-5. There was no change or  refactoring to the existing code. Yes, we needed to change some imports, but I think that's ok and not a refactoring.

Now we have duplicate imports [BeforeClass/BeforeAll](https://github.com/apache/ozone/pull/4236/files#diff-4275d85bf442cb352596248ced7651fcd5e3b09876bb6d2d80464cdf628a23fdR184-R185),  [AfterClass/AfterAll](https://github.com/apache/ozone/pull/4236/files#diff-4275d85bf442cb352596248ced7651fcd5e3b09876bb6d2d80464cdf628a23fdR244-R245) and [Test](https://github.com/apache/ozone/pull/4236/files#diff-4275d85bf442cb352596248ced7651fcd5e3b09876bb6d2d80464cdf628a23fdR126).
```
  @BeforeClass
  @BeforeAll
```
and 
```
  @AfterClass
  @AfterAll
```

If you think that having duplicate imports (the current situation) and clean up is not needed is ok, I'm fine you merging it.

Also if we don't do these type of update/upgrade one by one, we will never upgrade existing stuff unless we are forced which is rare in my opinion.

Reply:
Hi @hemantk-12, there is more than just changing imports.
`@Rule` from JUnit 4 should also be replaced (although it's just around 10 lines of change).
I can submit another PR right after this to do the migration.

Reply:
Not going to stretch it or argue any further. But Changing `@Rule` is one line change and `Assert.assertXX` is also one or two in IDE if you do a static import.

You already have a +1. So feel free to merge.

Reply:
It's not because of amount of work. It's because it makes commit history messy.

Reply:
It turns out there are some unexpected test logic change required in #4257.
See the `@Rule ExpectedException` part in this commit:  https://github.com/apache/ozone/pull/4257/commits/95a11d6a4c87cd772571e33f3493d200825975ef

Reply:
> But if you look at the original patch, these are the new tests which were added and I asked them to be in Junit-5.

@hemantk-12 Sorry for saying that, but this caused the problem.

Reply:
> if we don't do these type of update/upgrade one by one, we will never upgrade existing stuff unless we are forced which is rare in my opinion.

If you look at Ozone's history, it was @kaijchen who migrated lots of tests to JUnit5.

Reply:
Hi @adoroszlai, I think the best way is to merge #4257 first. So there won't be any JUnit5 related changes in this PR.